Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import WebKit
+
+struct TopTabsUX {
+    static let TopTabsViewHeight: CGFloat = 40
+    static let TopTabsBackgroundNormalColor = UIColor(red: 235/255, green: 235/255, blue: 235/255, alpha: 1)
+    static let TopTabsBackgroundPrivateColor = UIColor(red: 90/255, green: 90/255, blue: 90/255, alpha: 1)
+    static let TopTabsBackgroundNormalColorInactive = UIColor(red: 53/255, green: 53/255, blue: 53/255, alpha: 1)
+    static let TopTabsBackgroundPadding: CGFloat = 35
+    static let TopTabsBackgroundShadowWidth: CGFloat = 35
+    static let TabWidth: CGFloat = 180
+    static let CollectionViewPadding: CGFloat = 15
+    static let FaderPading: CGFloat = 5
+    static let BackgroundSeparatorLinePadding: CGFloat = 5
+    static let TabTitleWidth: CGFloat = 110
+    static let TabTitlePadding: CGFloat = 10
+}
+
+protocol TopTabsDelegate: class {
+    func topTabsPressTabs()
+    func topTabsPressNewTab()
+    func topTabsPressPrivateTab()
```

Comment:
Nit: Minor thing but the Cocoa APIs usually include a more precise timing of the events such as Will/Did to the method to clarify exactly when we are interacting with an element. I would just name these to `topTabsDidPressNewTab` if we called it after pressing the tab button.